### PR TITLE
Run remotedata tests only on cron job

### PR DESCRIPTION
**Pull request by dr-leo**
Note that behavior within the cron job can only be tested after merging into master.

### CODE REVIEW

**Comment by dr-leo** [comment]
While this PR addresses the suggestions made in #115 only insofar as remote-data dependent tests are run in cron jobs only, I would like to see this merged while leaving #115 open. We should add more specimen files first. In particular, some remote tests reveal bugs, eg in model. FacetType.

**Comment by khaeru** [comment]
First, as mentioned at #115, this requires no new file, nor any change to travis.yml—only proper use of existing [pytest hooks](https://docs.pytest.org/en/latest/reference.html#hooks) and [configuration](https://docs.pytest.org/en/latest/example/markers.html#scoped-marking).

A similar example is here: https://github.com/iiasa/message_ix/blob/c72f6aa4695da81fcb7932bcd7dd63f1cfff3ff9/tests/test_nightly.py#L15-L24

I will push some commits to this branch to show how that would work.

Second:

> We should add more specimen files first. In particular, some remote tests reveal bugs, eg in model. FacetType.

This is what I meant when I wrote, in #115:

> Certain variants of the SDMX-ML format are only tested via the remote tests.

etc.

It will be a lot of work to collect a full set of specimens. (I'm not even sure it is our task to do that—I feel like the standard should come with a full set.) Until then, I think the cron build should run daily, not weekly. That will at least make it possible to notice source changes sooner and respond to them.

**Comment by khaeru** [comment]
@dr-leo FYI, see the changes. My thinking here is:

- Some of the --remote-data tests *should* run on every Travis build. These test very basic functionality.
  - E.g. the snippets of code that are shown in the documentation. If these snippets stop working, then any user reading the documentation will not actually be able to use the advertised features. In this sense, they're critical and should be tested every time.
  - These are still fairly quick. The build now runs in under 2 minutes!
- If you run `pytest --remote-data` on your local system, the `pytestmark` condition evaluates to `skipif(False)`, i.e. *all* of the remote data tests will run. 
  - We need to do this because the remote-data tests are currently the only way to check whether the code covers all possible SDMX-ML data.
  - This also means when reviewing a PR, the Travis CI check is **no longer sufficient** per se. The reviewer **must** check out the branch locally and run with `--remote-data` to confirm all these tests pass, before approving/merging. Otherwise, we would accidentally merge code that breaks these tests, and not realize until the next cron build.

**Comment by dr-leo** [comment]
I agree that pytest Features are the better place to handle this. My Little script was too much of an ad-hoc hack.

Daily as opposed to Weekly cron jobs seem acceptable. I also agree that skipping all remote Tests except for cron jobs is second-best. That said, I would very much prefer a fall-back mechanism based on cached files instead of a failed download. I haven't looked at mock classes yet and cannot Judge their usefulness in this context. 

I feel strongly though that I have spent a disproportionate amount of time on the fragility of the test Suite. Instead, I would have preferred to Focus on the core where I see ample room for improvement.

Subjecting the success of a merge into master to the uptime and maintenance periods of a bunch of Servers around the world is a drag on the development process.  

Btw, the widespread use of xfails seems to testify that the battle against the fragility of the test Suite was lost Long ago. (Caveat: I haven't gone through the xfail-marked Tests to draw up a classification).

In summary, I would consider daily cron Jobs with full test execution, perhaps in combination with cached or otherwise stored data files  as a pragmatic temporary measure to free up scarce Resources to improve on the core. Such cached files would have to be managed. And I haven't researched how best to do this. 

To move Forward, I support merging this as a first step in the Right direction to be followed by others.

Final question for now: Why are there 3, not 4 checks on your commits?
